src/ripple/rpc/handlers/AccountObjects.cpp

```diff
@@ -0,0 +1,190 @@
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/rpc/impl/Tuning.h>
+#include <ripple/protocol/Indexes.h>
+
+namespace ripple {
+
+/** General RPC command that can retrieve objects in the account root.    
+    {
+      account: <account>|<account_public_key>
+      account_index: <integer> // optional, defaults to 0
+      ledger_hash: <string> // optional
+      ledger_index: <string | unsigned integer> // optional
+      limit: <integer> // optional
+      marker: <opaque> // optional, resume previous query
+    }
+*/
+Json::Value doAccountObjects (RPC::Context& context)
+{
+    auto const& params (context.params);
+    if (! params.isMember (jss::account))
+        return RPC::missing_field_error (jss::account);
+
+    Ledger::pointer ledger;
+    Json::Value result (RPC::lookupLedger (params, ledger, context.netOps));
+    if (ledger == nullptr)
+        return result;
+
+    RippleAddress rippleAddress;    
+    {
+        bool bIndex;
+        std::string const strIdent (params[jss::account].asString ());
+        int const iIndex (context.params.isMember (jss::account_index)
+            ? context.params[jss::account_index].asUInt () : 0);
+        Json::Value const jv(RPC::accountFromString(ledger, rippleAddress, bIndex,
+            strIdent, iIndex, false, context.netOps));
+        if (! jv.empty ())
+        {
+            for (Json::Value::const_iterator it (jv.begin ()); it != jv.end (); ++it)
+                result[it.memberName ()] = it.key ();
+
+            return result;
+        }
+    }
+
+    if (! ledger->hasAccount (rippleAddress))
+        return rpcError (rpcACT_NOT_FOUND);
+
+    unsigned int limit;
+    if (params.isMember (jss::limit))
+    {
+        auto const& jvLimit (params[jss::limit]);
+        if (! jvLimit.isIntegral ())
+            return RPC::expected_field_error (jss::limit, "unsigned integer");
+
+        limit = jvLimit.isUInt () ? jvLimit.asUInt () :
+            std::max (0, jvLimit.asInt ());
+
+        if (context.role != Role::ADMIN)
+        {
+            limit = std::max (RPC::Tuning::minObjectsPerRequest,
+                std::min (limit, RPC::Tuning::maxObjectsPerRequest));
+        }
+    }
+    else
+    {
+        limit = RPC::Tuning::defaultObjectsPerRequest;
+    }
+    
+    Account const& raAccount (rippleAddress.getAccountID ());
+    unsigned int reserve (limit);
+    uint256 startAfter;
+    std::uint64_t startHint (0);
+
+    if (params.isMember (jss::marker))
+    {
+        // We have a start point. Use limit - 1 from the result and use the
+        // very last one for the resume.
+        Json::Value const& marker (params[jss::marker]);
+
+        if (! marker.isString ())
+            return RPC::expected_field_error (jss::marker, "string");
+
+        startAfter.SetHex (marker.asString ());
+        SLE::pointer sleObj (ledger->getSLEi (startAfter));
+
+        if (sleObj == nullptr)
```

Comment:
I like to compare pointers to nullptr and booleans with ! (Meyers and others recommended)